### PR TITLE
fix: get correct env from npm config

### LINKED ISSUE #24556: Support lowercase environment variables to make `npm config` working.
### Current behavior

`npm config set CYPRESS_DOWNLOAD_PATH_TEMPLATE 'https://cdn.npmmirror.com/binaries/cypress/${version}/${platform}-${arch}/cypress.zip'` not working in current behavior.

```sh
npm config set CYPRESS_DOWNLOAD_PATH_TEMPLATE 'https://cdn.npmmirror.com/binaries/cypress/${version}/${platform}-${arch}/cypress.zip'
DEBUG='cypress:*' npm i cypress
```

You can see download is still `https://download.cypress.io/desktop/10.11.0?platform=linux&arch=x64`

### Desired behavior

`npm config set` will inject lowercase environment variables.

E.g: `npm config set CYPRESS_DOWNLOAD_PATH_TEMPLATE 'https://cdn.npmmirror.com/binaries/cypress/${version}/${platform}-${arch}/cypress.zip'` will inject:

```sh
npm_config_cypress_download_path_template=https://cdn.npmmirror.com/binaries/cypress/${version}/${platform}-${arch}/cypress.zip
```

So I think https://github.com/cypress-io/cypress/blob/d52e610c0036a6c37196883b1c270a2e93ec91ff/cli/lib/util.js#L532-L564

Should also support lowercase environment variables.

### Test code to reproduce

```sh
npm config set CYPRESS_DOWNLOAD_PATH_TEMPLATE 'https://cdn.npmmirror.com/binaries/cypress/${version}/${platform}-${arch}/cypress.zip'
DEBUG='cypress:*' npm i cypress
```

### Cypress Version

10.11.0

### Node version

18

### Operating System

Linux

### Debug Logs

```shell
(##################) ⠧ reify:cypress: timing reifyNode:node_modules/lodash Completed in 1475ms
> cypress@10.11.0 postinstall
> node index.js --exec install

  cypress:cli installing Cypress from NPM +0msyNode:node_modules/cypress Completed in 1495ms
  cypress:cli installing with options {} +0msfyNode:node_modules/cypress Completed in 1495ms
  cypress:cli detecting arch { osPlatform: 'linux', osArch: 'x64' } +0ms
  cypress:cli arm uname -m result: { stdout: 'x86_64' }  +34mses/cypress Completed in 1495ms
  cypress:cli version in package.json is 10.11.0, version to install is 10.11.0 +35ms
  cypress:cli Reading binary package.json from: /home/fengyu/.cache/Cypress/10.11.0/Cypress/resources/app/package.json +0ms
  cypress:cli no binary installed under cli version +16ms
  cypress:cli checking local file /tmp/tmp/10.11.0 cwd /tmp/tmp/node_modules/cypress +1ms
  cypress:cli preparing to download and unzip version  10.11.0 to path /home/fengyu/.cache/Cypress/10.11.0 +0ms
Installing Cypress (version: 10.11.0)


  cypress:cli needed Cypress version: 10.11.0 +0ms
  cypress:cli source url https://download.cypress.io/desktop/10.11.0?platform=linux&arch=x64 +1ms
  cypress:cli downloading cypress.zip to "/tmp/cypress-17897.zip" +0ms
  cypress:cli Downloading package {
  url: 'https://download.cypress.io/desktop/10.11.0?platform=linux&arch=x64',
  proxy: null,
  downloadDestination: '/tmp/cypress-17897.zip'
} +7ms
  cypress:cli expected file size 82 +1sng reifyNode:node_modules/cypress Completed in 1495ms
  cypress:cli redirect version: 10.11.0 +0ms
  cypress:cli redirect url: https://cdn.cypress.io/desktop/10.11.0/linux-x64/cypress.zip +0ms
  cypress:cli Downloading package {
  url: 'https://cdn.cypress.io/desktop/10.11.0/linux-x64/cypress.zip',
  proxy: null,
  downloadDestination: '/tmp/cypress-17897.zip'
} +1ms
  cypress:cli expected checksum c33b2a3eba0277da47274aff2e7a4e618531977355951d6f078ebd313d0a25d1807d6f5c30ccac19f0c292a1093f0494eb963e7e9a9db784fba41bd06b77567e +675ms
```
```


### Other

_No response_

### PR DESCRIPTION
**Pull request by abcfy2**
`npm config set VAR VAL` will inject `npm_config_var=val` environment variable. This commit will solve this issue

Closes: #24556

### CODE REVIEW

**Review by ryanthemanuel — CHANGES_REQUESTED** [review]
Can we add a test for this?

**Comment by abcfy2** [comment]
> Can we add a test for this?

I don't know how to add this test case. Because `npm config set FOO BAR`should execute before test.

If we add this command in unit test, then we should execute this test in sub process.

**Comment by abcfy2** [comment]
> Can we add a test for this?

I try to add a simple unit test for this. You can review again. Thanks.

**Review by ryanthemanuel — APPROVED** [review]

**Review by mjhenkes — APPROVED** [review]

**Comment by abcfy2** [comment]
Hi @marktnoonan . Please review my PR and let me know if I lost something. Thanks.

File: system-tests/lib/system-tests.ts [hunk 1]
```diff
@@ -629,6 +629,9 @@ const systemTests = {
       normalizeStdoutAvailableBrowsers: true,
       noExit: process.env.NO_EXIT,
       inspectBrk: process.env.CYPRESS_INSPECT_BRK,
+      env: {
+        CYPRESS_COMMERCIAL_RECOMMENDATIONS: '0',
+      },
```
Comment (mjhenkes) [inline system-tests/lib/system-tests.ts]:
```suggestion
```

File: .circleci/workflows.yml [hunk 2]
```diff
@@ -629,6 +629,7 @@ commands:
       - run:
           name: Run system tests
           command: |
+            CYPRESS_COMMERCIAL_RECOMMENDATIONS=0
```
Comment (mjhenkes) [inline .circleci/workflows.yml]:
```suggestion
```

File: .circleci/workflows.yml [hunk 3]
```diff
@@ -656,6 +657,7 @@ commands:
...
```
Comment (mjhenkes) [inline .circleci/workflows.yml]:
```suggestion
```

File: .circleci/workflows.yml [hunk 4]
```diff
@@ -1527,7 +1529,7 @@ jobs:
     steps:
       - restore_cached_workspace
       - run:
-          command: yarn workspace @tooling/system-tests test:ci "test/non_root*spec*" --browser electron
+          command: CYPRESS_COMMERCIAL_RECOMMENDATIONS=0 yarn workspace @tooling/system-tests test:ci "test/non_root*spec*" --browser electron
```
Comment (mjhenkes) [inline .circleci/workflows.yml]:
```suggestion
          command:  yarn workspace @tooling/system-tests test:ci "test/non_root*spec*" --browser electron
```

I need to fix these in develop for them to apply

File: cli/test/lib/util_spec.js [hunk 5]
```diff
@@ -543,6 +543,11 @@ describe('util', () => {
       expect(util.getEnv('CYPRESS_FOO')).to.eql('')
     })
 
+    it('npm config set should work', () => {
+      process.env.npm_config_cypress_foo_foo = 'bazz'
+      expect(util.getEnv('CYPRESS_FOO_FOO')).to.eql('bazz')
+    })
+
```
Comment (mjhenkes) [inline cli/test/lib/util_spec.js:550]:
```suggestion


```

File: cli/test/lib/util_spec.js [hunk 6]
```diff
@@ -543,6 +543,12 @@ describe('util', () => {
...
+
```
Comment (mjhenkes) [inline cli/test/lib/util_spec.js]:
```suggestion
```

**Comment by mjhenkes** [comment]
Yay a green build, @abcfy2 thanks for the PR and being patient while i got the build fixed!

**Pull request merged**